File: problems-adapt/0001-0100/0044_longest-integer-streak/solution_hash_set_streak.py
class Solution:
    def longestIntegerStreak(self, nums: list[int]) -> int:
        # The set collapses duplicates and makes membership an O(1) test.
        values = set(nums)
        best = 0
        for value in values:
            # Only a true run start (no value - 1 present) triggers a walk;
            # each maximal run has exactly one such start, which keeps the
            # nested loop linear: every element is touched at most twice.
            if value - 1 not in values:
                length = 1
                # Walk upward through the run without sorting anything.
                while value + length in values:
                    length += 1
                best = max(best, length)
        return best

File: problems-originals/0101-0200/0128_longest-consecutive-sequence-crawl/solution_hash_set_streak.rs
use std::collections::HashSet;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        // The set collapses duplicates and makes membership an O(1) test.
        let values: HashSet<i64> = nums.iter().map(|&value| value as i64).collect();
        let mut best: i32 = 0;
        for &value in &values {
            // Only a true run start (no value - 1 present) triggers a walk;
            // each maximal run has exactly one such start, which keeps the
            // nested loop linear: every element is touched at most twice.
            if !values.contains(&(value - 1)) {
                let mut length: i64 = 1;
                // Walk upward through the run without sorting anything.
                while values.contains(&(value + length)) {
                    length += 1;
                }
                best = best.max(length as i32);
            }
        }
        best
    }
}
